<launch>
    <!-- 上层控制节点 -->
    <arg name="sub_goal_tolerance_distance" default="1.0" doc="距离终点的最小容忍值，对静态地图没有用"/>
    <arg name="normal_zero_vel_segment" default="15" doc="速度赋0的分母(给控制用),整数"/>
    <arg name="collision_zero_vel_segment" default="1" doc="速度赋0的分母(障碍物避碰用),整数"/>
    <arg name="use_complex_lane" default="true" doc="是否对既有前进又有后退的路径处理"/>
    <arg name="use_gear" default="true" doc="是否加入到达目标点时的档位判断"/>
    <arg name="first_horizontal_distance" default="2.0" doc="第一个子目标点（横向距离）"/>
    <arg name="second_vertical_distance" default="10.0" doc="第二个子目标点（纵向距离）"/>
    <arg name="third_nearby_distance" default="5.0" doc="第三个子目标点（纵向距离）"/>

    <arg name="vehicle_length_coll" default="4.788"/>
    <arg name="vehicle_width_coll" default="2.198"/>
    <arg name="vehicle_cg2back_coll" default="1.367"/>
    <arg name="lookahead_distance" default="20.0"/>
    <arg name="waypoints_velocity_behav" default="1.0"/>

    <arg name="mode" default="1" doc="Scenario: MultiTrajPlanning"/>

    <arg name="loop_rate" default="100.0"/>

    <arg name="read_file_path" default="/home/nvidia/suv_car/src/mpc_traj_files/mpc_traj.txt"/>
    <arg name="save_file_path" default="/home/nvidia/suv_car/src/mpc_traj_files/mpc_traj.txt"/>

    <node name="behaviour_state_machine" pkg="behaviour_state_machine" type="behaviour_state_machine" output="screen" respawn="true">
        <param name="sub_goal_tolerance_distance" type="double" value="$(arg sub_goal_tolerance_distance)"/>
        <param name="normal_zero_vel_segment" type="int" value="$(arg normal_zero_vel_segment)"/>
        <param name="collision_zero_vel_segment" type="int" value="$(arg collision_zero_vel_segment)"/>
        <param name="use_complex_lane" type="bool" value="$(arg use_complex_lane)"/>
        <param name="use_gear" type="bool" value="$(arg use_gear)"/>
        <param name="first_horizontal_distance" type="double" value="$(arg first_horizontal_distance)"/>
        <param name="second_vertical_distance" type="double" value="$(arg second_vertical_distance)"/>
        <param name="third_nearby_distance" type="double" value="$(arg third_nearby_distance)"/>
        <param name="vehicle_length" type="double" value="$(arg vehicle_length_coll)"/>
        <param name="vehicle_width" type="double" value="$(arg vehicle_width_coll)"/>
        <param name="vehicle_cg2back" type="double" value="$(arg vehicle_cg2back_coll)"/>
        <param name="lookahead_distance" type="double" value="$(arg lookahead_distance)"/>
        <param name="waypoints_velocity" type="double" value="$(arg waypoints_velocity_behav)"/>
        <param name="mode" type="int" value="$(arg mode)"/>
        <param name="loop_rate" type="double" value="$(arg loop_rate)"/>
        <param name="read_file_path" type="string" value="$(arg read_file_path)"/>
        <param name="save_file_path" type="string" value="$(arg save_file_path)"/>

    </node>

</launch>